ddtrace/tracer/propagator.go

```diff
@@ -21,6 +21,16 @@ type Propagator interface {
 	Extract(carrier interface{}) (ddtrace.SpanContext, error)
 }
 
+// ExperimentalInjector implementations should be able to inject
+// SpanContexts into an implementation specific carrier. It is closely related to
+// Propagator except that it defines an experimental interface allowing more flexibility on
+// which keys are injected.
+// Note that this interface isn't meant to be public and used for other usages than internal ones.
+type ExperimentalInjector interface {
```

Comment:
I'm not hugely partial to the name `ExperimentalInjector` It seems like something that could be useful, so I'm not sure why to make it public but comment that it shouldn't be used otherwise.
Maybe a better name would be something like `WithOptionsInjector`?

Reply:
For what it's worth, I would have liked the new `Injector` to be private at this stage but since it's needed outside the `tracer` package, I had to make it public. 

But good points about the naming. I didn't like it either. `WithOptionsInjector` seems better.